lib/shopify_api/graphql/bulk_fetch.ex

```diff
@@ -0,0 +1,215 @@
+defmodule ShopifyAPI.GraphQL.BulkFetch do
+  alias ShopifyAPI.AuthToken
+
+  @max_poll_count 100
+  @polling_timeout_message "BulkFetch timed out before completion"
+  @polling_query """
+  {
+    currentBulkOperation {
+      id
+      status
+      errorCode
+      createdAt
+      completedAt
+      objectCount
+      fileSize
+      url
+      partialDataUrl
+    }
+  }
+  """
+
+  @doc """
+  ## Example
+      iex> prod_id = 10
+      iex> query = \"""
+        {
+          product(id: "gid://shopify/Product/\#{prod_id}") {
+            collections(first: 1) {
+              edges {
+                node {
+                  collection_id: id
+                  }
+                }
+              }
+            metafields(first: 1) {
+              edges {
+                node {
+                  key
+                  value
+                  metafield_id: id
+                }
+              }
+            }
+          }
+        }
+      \"""
+      iex> {:ok, token} = YourShopifyApp.ShopifyAPI.Shop.get_auth_token_from_slug("slug")
+      iex> ShopifyAPI.GraphQL.BulkFetch.process!(token, query)
+      {:ok,
+        [%{"collection_id" => "gid://shopify/Collection/xxx", ...}],
+      }
+  """
+  @spec process!(AuthToken.t(), String.t(), integer()) :: {:ok, list()} | {:error, any()}
```

Comment:
🍎 We probably shouldn't return a tuple here, Elixir convention says ! functions return the result and throw an exception if there is an error.